### DISCUSSION TITLE
Further functionality and customisability for `BeatmapAttributeText`'s attributes

### DISCUSSION THREAD

**Discussion by ikkue** [ppy/osu#21536] [Ideas]
Currently, the `BeatmapAttributeText` that was implemented with #21065 only displays the beatmap's information without any modification even if mods are applied to the current beatmap. There are three separate groups of attributes that, in my opinion, needs to be able to change with the mods being applied and which should be further customisable.

**1. `Length` and `BPM`**

- As @Feodor0090 has pointed out in https://github.com/ppy/osu/pull/21519#issuecomment-1339566068, "Info elements are supposed to show info, not gameplay state". That is the reason why #21519 was created in the first place separate from #21065. So my suggestion is to have further customisation for each attribute, such as `DrainLength` for `Length`, and `BPMRange` and `MostFrequentBPM` for `BPM`.

- Furthermore, both `Length` and `BPM` should change based on if Double Time or Nightcore is applied, as both apply an unchanging amount of multiplier throughout the whole beatmap, the same goes for Half Time and Daycore.

- As for mods that dynamically change the speed of the beatmap, like Wind Up or Wind Down, the **current** BPM during the map should already be handled via #21519 but the base `BPM` attribute, if present on the skin and effectively the screen, should switch to the new `BPMRange` format as mentioned above automatically.

**2. `Title` and `Artist`**

- This one is simple; they should be displayed in the original language if "Prefer metadata in original language" is selected in the settings.

**3. `CircleSize`, `HPDrain`, `Accuracy`, `ApproachRate`, and `StarRating`**

- This one is also relatively simple; if any mods, like Hard Rock, Difficulty Adjust, etc., affect the Circle Size (CS), HP Drain (HP), Accuracy/Overall Difficulty (OD), Approach Rate (AR), and/or Star Rating (SR) of the beatmap, the corresponding attributes should change to reflect what will be/is being played in the gameplay once it starts/has started.

**Comment by bdach** [comment]
let me say once more that i am hating the scope creep on this feature *so* much right now. this and https://github.com/ppy/osu/pull/21504
